### DISCUSSION TITLE
Next build fails when inside of a monorepo because of scss module typing.

### DISCUSSION THREAD

**Discussion by savager** [vercel/next.js#35459] [Help]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

    Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 21.1.0: Wed Oct 13 17:33:01 PDT 2021; root:xnu-8019.41.5~1/RELEASE_ARM64_T6000
    Binaries:
      Node: 16.13.0
      npm: 8.1.0
      Yarn: N/A
      pnpm: 6.32.3
    Relevant packages:
      next: 12.1.1-canary.15
      react: 17.0.2
      react-dom: 17.0.2

### Which example does this report relate to?

Any example, but I've recreated it with the blog

### What browser are you using? (if relevant)

not relevant

### How are you deploying your application? (if relevant)

not relevant 

### Describe the Bug

I'm sorry if this is a misunderstanding on my part, but I've tried this a few ways now to verify the issue.

So far I have tried, pnpm workspaces, yarn workspaces, nx, turborepo (both with default starter setup) all suffer this same issue.

The simplest example would be:

```npx create-turbo@latest```

I chose pnpm but I doubt it matters. finish setup.

Then add a single scss module with import to the index.tsx file.

```import { example } from './index.module.scss'```

![image](https://user-images.githubusercontent.com/1455337/159032549-9709a0b2-0b45-4af8-b9e9-07e8d2e92a0c.png)

This will lint as an error right away:

![image](https://user-images.githubusercontent.com/1455337/159031644-d485864b-7438-45f7-b844-6b84a371760a.png)

It will also fail the build:

![image](https://user-images.githubusercontent.com/1455337/159031731-b7f79c80-fa18-442b-aed5-27d1d66f4412.png)

I can get rid of the _linting_ error by installing the package "typescript-plugin-css-modules", adding a .module.scss types, and referencing them in the tsconfig file. I also needed to add to my vscode settings.json:

![image](https://user-images.githubusercontent.com/1455337/159032176-94630689-73fd-4049-aefb-fe8dd468c718.png)

But even when the linting is fixed, the next build still fails without adding "ignoreBuildErrors: true" to my next.config.

I've also tried installing next at the root of the project to no resolution.

### Expected Behavior

I would like the next boilerplate to include linting and building support for (s)css modules when nested. I'm hoping this is a quick fix and someone can point me in the right direction and this will act to help others. Thanks!

### To Reproduce

Included in the description.

**Comment by savager** [comment]
The types I've added: 
![image](https://user-images.githubusercontent.com/1455337/159038331-4e68e877-175f-4c96-9652-810b30c2c781.png)

**Comment by savager** [comment]
I should probably close this, I didn't realize these approaches were all using this package: https://github.com/martpie/next-transpile-modules

Problem might lie there.

**Comment by balazsorban44** [comment]
Make sure that the file where you declared your modules is included in the `tsconfig.json` file's `include` array. You might need to restart the VSCode TypeScript Server (re-opening the editor should also work I guess)

**Reply by savager** [reply]
I did thanks. I believe it could also be this, no named exports https://github.com/mrmckeb/typescript-plugin-css-modules/issues/16

because using it with ```import * as styles``` works

![image](https://user-images.githubusercontent.com/1455337/159135743-8c5eb4f7-19b8-4bd1-be13-e39760d3dbfc.png)

ignore me verifying destructuring is working :)

**Comment by savager** [comment]
I don't even know if nextjs supports named exports from scss modules tbh. I just imported them the normal way and all is well.
